### PR TITLE
Sn testing/reading list

### LINKED ISSUE #39: Test ReadingListBook component

### PR DESCRIPTION
**Pull request by NakiNorton**
#### What's this PR do?
- Tests that the ReadingList and ReadingListBook components are being rendered correctly
- Tests that the reducer and actions related to the ReadingList functionality are working as expected
- Connects the ReadingList component to the store and refactor so it's now getting the readingList directly from store (instead of being passed in from Books)
- Tests that App is rendering child components correctly

Type of change made:
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Styling- no new features
- [X] Testing

#### Where should the reviewer start?
- ReadingList.test file

#### Why is this change required? What problem does it solve?
- Ensures that components are rendering correctly and the store is connected properly

#### Were there any challenges that arose while implementing this feature? If so, how were they addressed?
-  It was challenging to figure out how to mock the store and test actions. Googling helped me to work this out.

#### Any background context you want to provide?
- While testing the ReadingList component I realized that we were passing in readingList from Books, even though it was in the store and didn't need to be passed in anymore. This led me to connecting the ReadingList to the store.
- I added in two tests for App component not related to the branch name. 

#### What are the relevant tickets?
- resolves #39 #38 #40

#### Screenshots (if appropriate)
-  n/a

#### Questions:
- Should we put all of the tests for 'actions' in one file, and tests for 'reducers' in one file? Or do you think it makes more sense to have the tests in separate files to mirror the action/reducer file structure (where they're all in their own separate file)?

### CODE REVIEW

File: src/Components/App/App.test.js [hunk 1]
```diff
@@ -1,9 +1,89 @@
...
+    render(
+      <Provider store={store}>
+        <MemoryRouter>
+          <App />
+        </MemoryRouter>
+      </Provider>
+    )
+  })
+
+  it('should render the Book component when app loads', async () => {
```
Comment (NakiNorton) [inline src/Components/App/App.test.js:59]: This should say 'should render the Books Component......' not Book

File: src/Components/ReadingListBook/ReadingListBook.test.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            title={'WHERE THE CRAWDADS SING'}
+            image={'https://s1.nyt.com/du/books/images/9780735219090.jpg'}
+            id={'015489213'}
+            key={'015489213'}
+          />
+        </MemoryRouter>
+      </Provider>
+    )
+  })
+
```
Comment (amykarnaze) [inline src/Components/ReadingListBook/ReadingListBook.test.js:46]: Nice job adding this content filled before each!

**Review by amykarnaze — APPROVED** [review]
Great job implementing redux in testing!

**Comment by NicoleLatifi** [comment]
I think either file structure you mentioned would be fine. Personally, I like the idea of mirroring the actions and reducers file structure for their tests and keeping all the tests in separate files.

File: src/Components/Books/Books.js [hunk 3]
```diff
@@ -72,12 +72,12 @@ class Books extends Component {
...
 
   render() {
-    const { books, readingList } = this.props
+    const { books } = this.props
     let bookCards = this.displayBooks()
     return (
       <Switch>
         <Route exact path='/favorites' render={() =>
-          <ReadingList toReadList={readingList} /> } 
+          <ReadingList /> }
```
Comment (NicoleLatifi) [inline src/Components/Books/Books.js:79]: **praise:** Good catch that we don't need this anymore!

**Review by NicoleLatifi — APPROVED** [review]
Thank you for all of your work on these tests!

**Comment by kip-west** [comment]
Thanks for all your work on getting the Redux testing solid!

I think mirroring the file structure of reducers/actions makes sense as well!

**Review by kip-west — APPROVED** [review]

**Pull request merged**